plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineExecutionService.java

```diff
@@ -43,7 +43,7 @@ public PipelineExecutionService(PipelineStore store, ThreadPool threadPool) {
         this.threadPool = threadPool;
     }
 
-    public void execute(IndexRequest indexRequest, String pipelineId, ActionListener<IngestDocument> listener) {
+    public void execute(IndexRequest indexRequest, String pipelineId, ActionListener<Void> listener) {
```

Comment:
sounds good to me, not a big fan of my change either. I just wanted to make sure that nobody relies on IngestDocument though here, cause it gets modified before the document is set back to the IndexRequest.